resources/user.rb

```diff
@@ -74,6 +80,14 @@ def load_current_value
     execute "Delete #{new_resource.name}" do
       command "smbpasswd -x #{new_resource.name}"
     end
-    new_resource.updated_by_last_action(true)
+  end
+end
+
+action_class.class_eval do
+  require 'mixlib/shellout'
+  def generate_system_password
+    system_password = Mixlib::ShellOut.new("/usr/bin/openssl passwd -1 #{new_resource.password}").run_command.stdout.strip
```

Comment:
this might be more efficient and negate needing to install openssl. it works very well on my other cookbooks.
system_password = new_resource.password.crypt('$6$' + SecureRandom.random_number(36 ** 8 ).to_s(36))